src/Features/CSharp/Portable/RemoveAsyncModifier/CSharpRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,118 @@
+        public CSharpRemoveAsyncModifierCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(CS1998);
+
+        protected override bool IsAsyncSupportingFunctionSyntax(SyntaxNode node)
+            => node.IsAsyncSupportingFunctionSyntax();
+
+        protected override bool CanFix(KnownTypes knownTypes, IMethodSymbol methodSymbolOpt)
+        => !methodSymbolOpt.ReturnType.OriginalDefinition.Equals(knownTypes._iAsyncEnumerableOfTTypeOpt) &&
+           !methodSymbolOpt.ReturnType.OriginalDefinition.Equals(knownTypes._iAsyncEnumeratorOfTTypeOpt);
+
+        protected override SyntaxNode ChangeReturnStatements(SyntaxNode node, SyntaxGenerator generator, KnownTypes knownTypes)
+        {
+            var converter = new ConvertReturnsToTaskRewriter(generator, knownTypes._taskType);
+
+            return node switch
+            {
+                MethodDeclarationSyntax method => FixMethodReturns(method, converter),
+                LocalFunctionStatementSyntax localFunction => FixLocalFunctionReturns(localFunction, converter),
+                AnonymousFunctionExpressionSyntax method => FixAnonymousFunctionReturns(method, converter),
+                _ => node,
+            };
+        }
+
+        private static AnonymousFunctionExpressionSyntax FixAnonymousFunctionReturns(AnonymousFunctionExpressionSyntax lambda, ConvertReturnsToTaskRewriter converter)
+        {
+            if (lambda.Block != null)
+            {
+                return lambda.WithBlock((BlockSyntax)converter.Visit(lambda.Block));
+            }
+            else
+            {
+                return lambda.WithExpressionBody(converter.WrapWithTaskFromResult(lambda.ExpressionBody));
+            }
+        }
+
+        private static MethodDeclarationSyntax FixMethodReturns(MethodDeclarationSyntax method, ConvertReturnsToTaskRewriter converter)
+        {
+            if (method.Body != null)
+            {
+                return method.WithBody((BlockSyntax)converter.Visit(method.Body));
+            }
+            else
+            {
+                return method.WithExpressionBody((ArrowExpressionClauseSyntax)converter.Visit(method.ExpressionBody));
+            }
+        }
+
+        private static LocalFunctionStatementSyntax FixLocalFunctionReturns(LocalFunctionStatementSyntax localFunction, ConvertReturnsToTaskRewriter converter)
+        {
+            if (localFunction.Body != null)
+            {
+                return localFunction.WithBody((BlockSyntax)converter.Visit(localFunction.Body));
+            }
+            else
+            {
+                return localFunction.WithExpressionBody((ArrowExpressionClauseSyntax)converter.Visit(localFunction.ExpressionBody));
+            }
+        }
+
+        protected override SyntaxNode RemoveAsyncModifier(IMethodSymbol methodSymbolOpt, SyntaxNode node, KnownTypes knownTypes) => node switch
+        {
```

Comment:
don't write this way (the braces look like method braces).  Instead do:
```suggestion
    => node switch
    {
    };
```

Then it's very clear that hte method is an expr-bodied member.